bin/generate

```diff
@@ -26,13 +29,20 @@ def fetch(exercise):
             msg = 'Unexpected error, please try again later.'
         raise Exception(msg)
 
-
 def load(exercise):
-    root = Path(__file__).absolute().parent.parent.parent
-    path = root / 'problem-specifications/exercises'
+    path = smldir().parent / 'problem-specifications' / 'exercises'
     with (path / exercise / 'canonical-data.json').open() as f:
         return json.loads(f.read())
 
+def get_canonical_data(exercise, problem_specs_source='remote'):
+    """Get canonical-data of the exercise as a json string."""
+    if problem_specs_source == 'remote':
+        return fetch(exercise)
+    elif problem_specs_source == 'local':
+        return load(exercise)
+    else:
+        raise Exception("Unknown source for problem-specifications: '{}'."
```

Comment:
Is the only reason for this exception that the source is unknown, or can it also be that it is unreachable?  Can we give a better and more specific exception message?  Perhaps directive?  (The difference between unreachable because of no network versus the resource was contacted, but what was expected to be found in the location specified was not there.)

Reply:
We might also suggest that, if it was the remote that failed, that we could opt to use the local, with this message.

Reply:
At least *this* exception should be correct, because control flow can only reach it if `problem_specs_source` has not one of the two known values. The `fetch` and `load` functions (which I essentially left as they were) can throw their own exceptions, I don't catch them. Or did I miss something?

You are probably right that it would be a good idea to provide the user with some more directives. I actually considered it by myself but didn't do it eventually since the script appears to me very "basic" / not very sophisticated. I also do not really like the error handling (but I don't want to modify it too much in this PR). So I went for minimalist solutions. Also my python is a little bit rusty at the moment.

I will provide an upgrade with your suggestions somehow incorporated soon. I put the PR to draft again.